src/oadp_resources/storage_class/storage_class.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+
+from ocp_resources.resource import Resource
+
+
+class StorageClass(Resource):
+    """
+    StorageClass object.
+    """
+
+    api_group = Resource.ApiGroup.STORAGE_K8S_IO
+    def __init__(
+        self,
+        name=None,
+        client=None,
+        provisioner=None,
+        volume_binding_mode=None,
+        teardown=True,
+        yaml_file=None,
+        **kwargs,
+    ):
+        super().__init__(
+            client=client,
+            name=name,
+            teardown=teardown,
+            yaml_file=yaml_file,
+            **kwargs,
+        )
+        self.provisioner = provisioner
+        self.volume_binding_mode = volume_binding_mode
+
+    def to_dict(self):
+        if not self.res:
+            super().to_dict()
+
+        if not self.yaml_file and self.provisioner:
+            self.add_provisioner(
+                provisioner=self.provisioner,
+                volume_binding_mode=self.volume_binding_mode,
+            )
+
+    def add_provisioner(self,provisioner=None, volume_binding_mode=None):
+        if not self.res:
+            super().to_dict()
+        self.res["provisioner"]=provisioner
+        if volume_binding_mode:
+            self.res["volumeBindingMode"]=volume_binding_mode
```

Comment:
Is this really needed if we set it with the constructor?
also, the name is confusing,  
it is setting a provider, not adding one.
wdyt?

Reply:
maybe change the function name to set (it is setting the provider, not adding ine, correct?)

Reply:
haa correct Amos I will update the commit in a moment

Reply:
Modified Amos please give it a check